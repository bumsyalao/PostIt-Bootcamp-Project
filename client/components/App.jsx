 /*./client/components/App.js */

import React from 'react';

class App extends React.Component{
  render() {
    return (
      <div style ={{textAlign: 'center'}}>
        <h1>Welcome to Post-It</h1>
      </div>);
  }
}