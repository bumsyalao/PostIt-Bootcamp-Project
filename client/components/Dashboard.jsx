import React from 'react';
import { Link } from 'react-router-dom';

const Dashboard = () => (
        <div className="section no-pad-bot" id="index-banner">
          <div className="container">
            <br></br>
            <div className="center">
            <img src ="https://www.embracingspirituality.com/wp-content/uploads/wpforo/default_attachments/1489579326-welcome.jpg"/>
            </div>
            <div className="row center">
              <h3 className="header col s12 light">Share Anything, With Anyone!</h3>
            </div>
            <div className="row center">
              <a href="register1.html" id="download-button" className="btn-large waves-effect waves-light red lighten-2">
                <Link to="/signup">Get Started</Link> </a>
            </div>
            <br></br>

          </div>
        </div>
    );

export default Dashboard;

