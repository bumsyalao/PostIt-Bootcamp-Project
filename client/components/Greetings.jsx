import React from 'react';

class Greetings extends React.Component {
   render() { 
    return (
        <div className="section no-pad-bot" id="index-banner">
          <div className="container">
            <br></br>
            <div className="center">
            <img src ="https://www.embracingspirituality.com/wp-content/uploads/wpforo/default_attachments/1489579326-welcome.jpg"/>
            </div>
            <div className="row center">
              <h3 className="header col s12 light">Share Anything, With Anyone!</h3>
            </div>
            <div className="row center">
              <a href="register1.html" id="download-button" className="btn-large waves-effect waves-light red lighten-2">Get Started</a>
            </div>
            <br></br>
          </div>
        </div>

    );
 }
}

export default Greetings;