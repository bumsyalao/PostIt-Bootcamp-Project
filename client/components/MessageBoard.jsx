// import React, { Component } from 'react';

// class MessageBoard extends Component {
//   render(){
//       if (localStorage.getItem("user") === null) {
//   // eslint-disable-next-line
//   this.props.history.push(`/siginin`);
//   return;
//   } else {
//     return (

//     )
//   }
// }