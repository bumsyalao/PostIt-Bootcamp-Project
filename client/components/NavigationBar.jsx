import React from 'react';
import { Link } from 'react-router-dom';

const NavigationBar = () => (
      <nav>
          <div className="nav-wrapper container">
              <Link to="/">
              <a id="logo-container" className="brand-logo">POST-IT</a>
              </Link>
            <ul className="right">
              <li><a href="login.html">Login</a></li>
            </ul>
            <ul className="right">
              <li><Link to="/signup">Register</Link></li>
            </ul>
            <ul id="nav-mobile" className="side-nav">
              <li><a href="#">Navbar Link</a></li>
            </ul>
            <a href="#" data-activates="nav-mobile" className="button-collapse"><i className="mdi-navigation-menu"></i></a>
          </div>
        </nav>
    );

export default NavigationBar;