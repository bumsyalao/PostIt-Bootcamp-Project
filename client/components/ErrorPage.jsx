import React from 'react';
import NavigationBar from './NavigationBar';


const ErrorPage = () => (
  <div>
    <NavigationBar />
    <h4 className="error-page"> Error 404: Page Not Found </h4>
  </div>

);

export default ErrorPage;
