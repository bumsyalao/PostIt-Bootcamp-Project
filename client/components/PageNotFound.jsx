import React from 'react';
import NavigationBar from './NavigationBar';

const PageNotFound = () => (
  <div>
  <NavigationBar/>
    <h4 className="error-page"> Error 404: Page Not Found </h4>
  </div>

);

export default PageNotFound;
