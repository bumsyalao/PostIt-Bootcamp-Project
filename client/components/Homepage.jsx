import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import CreateGroup from './Group/CreateGroup';
import Sidebar from './Sidebar';
import ListGroup from './Group/ListGroup';
import GroupChat from './Group/GroupChat';
import UsersPage from '../components/Users/UsersPage';
import UserProfile from '../components/Users/UserProfile';
import Welcomepage from './Welcomepage';
import PageNotFound from './PageNotFound';
import SearchUser from '../components/Group/SearchUser';
import GroupUsers from '../components/Group/GroupUsers';

/**
 * @class Homepage
 * @extends {React.Component}
 */
export class Homepage extends React.Component {

  /**
   *
   * Check if user is authenticated
   * Toast access message
   * Intialize materialize collapsible
   *
   * @memberOf Homepage
   */
  componentDidMount() {
    $('.collapsible').collapsible();
    if (!this.props.access.isAuthenticated) {
      this.props.history.push('/signin');
      Materialize.toast('Please SignIn or Register', 5000, 'red');
    }
  }

  /**
   * Renders Homepage
   * @returns Homepage
   * @memberOf Homepage
   */
  render() {
    return (
      <div>
        <Sidebar />
        <div className="homepage">

          <Switch>
            <Route
              path={`${this.props.match.url}/welcome-page`}
              component={Welcomepage}
            />
            <Route
              path={`${this.props.match.url}/create-group`}
              component={CreateGroup}
            />
            <Route
              path={`${this.props.match.url}/groups`}
              component={ListGroup}
            />
            <Route
              path={`${this.props.match.url}/view-group/:groupId`}
              component={GroupChat}
            />
            <Route
              path={`${this.props.match.url}/search-users`}
              component={UsersPage}
            />
            <Route
              path={`${this.props.match.url}/user-profile`}
              component={UserProfile}
            />
            <Route
              path={`${this.props.match.url}/group/:groupId/add-user`}
              component={SearchUser}
            />
            <Route
              path={`${this.props.match.url}/group/:groupid/group-users`}
              component={GroupUsers}
            />
            <Route component={PageNotFound}/>
          </Switch>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    access: state.access
  }
);

export default connect(mapStateToProps, null)(withRouter(Homepage));

