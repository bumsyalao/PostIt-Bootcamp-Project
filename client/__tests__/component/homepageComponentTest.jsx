/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import { Homepage } from '../../components/Homepage';

configure({ adapter: new adapter() });

describe('Hompeage component', () => {
  const spy = jest.fn();
  global.Materialize = { toast: jest.fn() };
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    match: {
      path: ''
    },
    history: {
      push: spy
    },
    access: {
      isAuthenticated: false
    },
  };
  const component = shallow(
    <Homepage {...props} />
  );

  it('should match the Dashboard component snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
