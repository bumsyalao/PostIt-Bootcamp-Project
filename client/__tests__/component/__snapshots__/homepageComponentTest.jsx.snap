// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hompeage component should match the Dashboard component snapshot 1`] = `
<div>
  <Connect(withRouter(Sidebar)) />
  <div
    className="homepage"
  >
    <Switch>
      <Route
        component={[Function]}
        path="undefined/welcome-page"
      />
      <Route
        component={[Function]}
        path="undefined/create-group"
      />
      <Route
        component={[Function]}
        path="undefined/groups"
      />
      <Route
        component={[Function]}
        path="undefined/view-group/:groupId"
      />
      <Route
        component={[Function]}
        path="undefined/search-users"
      />
      <Route
        component={[Function]}
        path="undefined/user-profile"
      />
      <Route
        component={[Function]}
        path="undefined/group/:groupId/add-user"
      />
      <Route
        component={[Function]}
        path="undefined/group/:groupid/group-users"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</div>
`;
